projects/ngx-tour-core/src/lib/tour-hotkey-listener.component.ts

```diff
@@ -26,10 +26,13 @@ export class TourHotkeyListenerComponent {
   public onArrowRightKey(): void {
     if (
       this.tourService.getStatus() === TourState.ON &&
-      this.tourService.hasNext(this.tourService.currentStep) &&
       this.tourService.isHotkeysEnabled()
     ) {
-      this.tourService.next();
+      if (this.tourService.hasNext(this.tourService.currentStep)) {
+        this.tourService.next();
+      } else if (!this.tourService.hasNext(this.tourService.currentStep)) {
```

Comment:
There is no point in checking the same thing twice, could be just `else` instead of `else if`.
Anyway, I don't think the tour should end when arrow key is pressed. Tour ends on `Esc` key.